core/src/main/java/org/apache/iceberg/RewriteTablePathUtil.java

```diff
@@ -354,7 +354,10 @@ private static RewriteResult<DataFile> writeDataFileEntry(
     DataFile newDataFile =
         DataFiles.builder(spec).copy(entry.file()).withPath(targetDataFilePath).build();
     appendEntryWithFile(entry, writer, newDataFile);
-    result.copyPlan().add(Pair.of(sourceDataFilePath, newDataFile.location()));
+    // Keep non-live entry but exclude deleted data files as part of copyPlan
```

Comment:
Nit: keep deleted data file entries but exclude them from copyPlan

(same for the following few comments)

My thought is, negative statements always are a bit harder to parse (non-live).

Reply:
Sounds good, updated as suggested for all 3 comments